src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -450,6 +452,23 @@ SpecialType.System_Collections_Generic_IReadOnlyCollection_T or
             return _factory.Convert(collectionType, arrayOrList);
         }
 
+        private BoundExpression VisitDictionaryInterfaceCollectionExpression(BoundCollectionExpression node)
+        {
+            Debug.Assert(!_inExpressionLambda);
+            Debug.Assert(_factory.ModuleBuilderOpt is { });
+            Debug.Assert(_diagnostics.DiagnosticBag is { });
+            Debug.Assert(node.Type is NamedTypeSymbol);
+            Debug.Assert(node.CollectionCreation is null);
+            Debug.Assert(node.Placeholder is null);
+
+            var collectionType = (NamedTypeSymbol)node.Type;
+            var typeArguments = collectionType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics;
+            var elements = node.Elements;
+            // PROTOTYPE: Create a custom interface implementation for IReadOnlyDictionary<K, V> to enforce immutability?
+            var collection = CreateAndPopulateDictionary(node, typeArguments, elements);
+            return _factory.Convert(collectionType, collection);
```

Comment:
Would this handle the situation where I use a custom `Dictionary` type that doesn't implement `IDictionary` interface? We may not need to address if we're planning to eliminate usage of `Dictionary` though